### ISSUE TITLE
Crash importing firehose delivery stream

### ISSUE THREAD

**Issue by grahamj** [terraform-providers/terraform-provider-aws#3833]
### Terraform Version
Terraform v0.11.4
+ provider.aws v1.11.0

### Affected Resource(s)
- aws_kinesis_firehose_delivery_stream

### Terraform Configuration Files
```hcl
terraform {
  backend "s3" {}
}

data "terraform_remote_state" "common" {
  # Snipped
}

resource "aws_kinesis_firehose_delivery_stream" "import" {
  # Stuff
}
```

### Panic Output
```
panic: runtime error: index out of range
2018-03-19T11:05:50.631-0400 [DEBUG] plugin.terraform-provider-aws_v1.11.0_x4:
2018-03-19T11:05:50.631-0400 [DEBUG] plugin.terraform-provider-aws_v1.11.0_x4: goroutine 102 [running]:
2018-03-19T11:05:50.631-0400 [DEBUG] plugin.terraform-provider-aws_v1.11.0_x4: github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsKinesisFirehoseDeliveryStream.func1(0xc4202c05b0, 0x3230a80, 0xc420092a00, 0x24, 0xc4202587a8, 0xc420294001, 0x316e8a0, 0xc4205c99c0)
2018-03-19T11:05:50.631-0400 [DEBUG] plugin.terraform-provider-aws_v1.11.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_kinesis_firehose_delivery_stream.go:383 +0x275
2018-03-19T11:05:50.631-0400 [DEBUG] plugin.terraform-provider-aws_v1.11.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).ImportState(0xc420203f80, 0xc420903770, 0xc4205d1140, 0x36, 0x10be37f, 0x2d7d140, 0x2d7d140, 0xc420064301, 0x6172b20)
2018-03-19T11:05:50.631-0400 [DEBUG] plugin.terraform-provider-aws_v1.11.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:365 +0x322
2018-03-19T11:05:50.631-0400 [DEBUG] plugin.terraform-provider-aws_v1.11.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).ImportState(0xc420148ac0, 0xc4205c97c0, 0xc4205c99a0, 0x0, 0x0)
2018-03-19T11:05:50.631-0400 [DEBUG] plugin.terraform-provider-aws_v1.11.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:560 +0x5d
2018-03-19T11:05:50.631-0400 [DEBUG] plugin.terraform-provider-aws_v1.11.0_x4: reflect.Value.call(0xc420304900, 0xc420512118, 0x13, 0x364b02f, 0x4, 0xc4205faf20, 0x3, 0x3, 0x8, 0xc4205f2e80, ...)
2018-03-19T11:05:50.631-0400 [DEBUG] plugin.terraform-provider-aws_v1.11.0_x4: 	/usr/local/go/src/reflect/value.go:434 +0x905
2018-03-19T11:05:50.631-0400 [DEBUG] plugin.terraform-provider-aws_v1.11.0_x4: reflect.Value.Call(0xc420304900, 0xc420512118, 0x13, 0xc4205f2f20, 0x3, 0x3, 0xc4205f2f20, 0xc4205f2f28, 0xc4205ca900)
2018-03-19T11:05:50.631-0400 [DEBUG] plugin.terraform-provider-aws_v1.11.0_x4: 	/usr/local/go/src/reflect/value.go:302 +0xa4
2018-03-19T11:05:50.631-0400 [DEBUG] plugin.terraform-provider-aws_v1.11.0_x4: net/rpc.(*service).call(0xc42037e140, 0xc420364230, 0xc420014898, 0xc420386780, 0xc420532460, 0x2d7d100, 0xc4205c97c0, 0x16, 0x2d7d140, 0xc4205c99a0, ...)
2018-03-19T11:05:50.631-0400 [DEBUG] plugin.terraform-provider-aws_v1.11.0_x4: 	/usr/local/go/src/net/rpc/server.go:381 +0x142
2018-03-19T11:05:50.631-0400 [DEBUG] plugin.terraform-provider-aws_v1.11.0_x4: created by net/rpc.(*Server).ServeCodec
2018-03-19T11:05:50.631-0400 [DEBUG] plugin.terraform-provider-aws_v1.11.0_x4: 	/usr/local/go/src/net/rpc/server.go:475 +0x36b
2018/03/19 11:05:50 [ERROR] root: eval: *terraform.EvalImportState, err: import aws_kinesis_firehose_delivery_stream.import (id: arn:aws:firehose:us-east1:XXXXXXXXX:XXXXXXXXX): unexpected EOF
2018/03/19 11:05:50 [ERROR] root: eval: *terraform.EvalSequence, err: import aws_kinesis_firehose_delivery_stream.import (id: arn:aws:firehose:us-east1:XXXXXXXXX:XXXXXXXXX): unexpected EOF
2018/03/19 11:05:50 [TRACE] [walkImport] Exiting eval tree: aws_kinesis_firehose_delivery_stream.import (import id: arn:aws:firehose:us-east1:XXXXXXXXX:XXXXXXXXX)
2018/03/19 11:05:50 [TRACE] dag/walk: upstream errored, not walking "provider.aws (close)"
2018/03/19 11:05:50 [TRACE] dag/walk: upstream errored, not walking "root"
2018/03/19 11:05:50 [DEBUG] plugin: waiting for all plugin processes to complete...
2018-03-19T11:05:50.634-0400 [WARN ] plugin: error closing client during Kill: err="connection is shut down"
2018-03-19T11:05:50.638-0400 [DEBUG] plugin: plugin process exited: path=blah/.terraform/plugins/darwin_amd64/terraform-provider-aws_v1.11.0_x4
```

### Command
`AWS_PROFILE=XXX terraform import aws_kinesis_firehose_delivery_stream.import arn:aws:firehose:us-east1:XXXXXXXXX:XXXXX`

Sorry, I don't have time right now to sanitize the crash.log but if you really need it just shout. Hopefully the above should point you in the right direction.

**Comment by bflad** [comment]
@grahamj its crashing due to a missing `/` in the last section of ARN you provided. Can you double check it matches the expected [stream ARN format](https://www.terraform.io/docs/providers/aws/r/kinesis_firehose_delivery_stream.html#import) or if your ARN for some reason does not match this format?

`arn:PARTITION:firehose:REGION:ACCOUNTID:deliverystream/example`

In the meantime, I'll submit a PR that correctly returns the error instead of the panic.

**Comment by bflad** [comment]
This PR will provide an error message instead of panic on import ARNs missing `deliverystream/`: #3834

**Comment by bflad** [comment]
The fix for the `aws_kinesis_firehose_delivery_stream` import to correctly throw an error instead of panic has been merged into master and will release with v1.12.0 of the AWS provider, likely later this week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.12.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1120-march-23-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by grahamj** [comment]
Oops I guess that was my bad providing an invalid ARN. An improved error message will certainly be helpful next time. Thanks!
